### ISSUE TITLE
Add filtering options to new submissions list

### ISSUE THREAD

**Issue by NateWr** [pkp/pkp-lib#2612]
Initial testing of the new submissions list indicates strong demand for additional filtering options:

### High Priority
- [ ] By "what I need to do now" categories. Details: https://github.com/pkp/pkp-lib/issues/2612#issuecomment-431791354. Discussion: ([All reviews in](https://github.com/pkp/pkp-lib/issues/4102), [Revisions uploaded](https://github.com/pkp/pkp-lib/issues/4102)) (now filed at https://github.com/pkp/pkp-lib/issues/7488)
- [ ] Disambiguate overdue review acceptance and review completion See https://github.com/pkp/pkp-lib/issues/4102

### Requested
- [ ] Overdue by X days. See https://github.com/pkp/pkp-lib/issues/7489
- [ ] Originally submitted more than X days ago. See https://github.com/pkp/pkp-lib/issues/3397
- [ ] Filter by language. See https://github.com/pkp/pkp-lib/issues/5263
- [ ] Filter archived submissions by Declined/Published.
- [x] Filter by category. See https://github.com/pkp/pkp-lib/issues/6809.

### Done
- [x] By stage in the workflow
- [x] By section
- [x] By assigned editor (only Journal Manager can access). Filed at https://github.com/pkp/pkp-lib/issues/5982
- [x] By assigned issue. Filed at https://github.com/pkp/pkp-lib/issues/6054. See https://github.com/pkp/pkp-lib/issues/2612#issuecomment-531341541 (The API already supports getting by issue ID: https://github.com/pkp/ojs/blob/master/classes/services/SubmissionService.inc.php#L49)
- [x] Last activity more than X days ago. Active work: https://github.com/pkp/pkp-lib/issues/4168#issuecomment-431195503. Comments: https://github.com/pkp/pkp-lib/issues/3604#issuecomment-386579636, https://github.com/pkp/pkp-lib/issues/4102

**Comment by NateWr** [comment]
PRs:
https://github.com/pkp/pkp-lib/pull/2704
https://github.com/pkp/ojs/pull/1490
https://github.com/pkp/omp/pull/428

Filtering is available on all the main submission lists for journal managers, subeditors and assistants.

![submission-list-filtering](https://user-images.githubusercontent.com/2306629/29220471-077b26c8-7eb3-11e7-9159-aea4b2583446.gif)

A couple questions for @jmacgreg and @stranack (and whoever):

1. Does this look like it covers the requests we got for filtering?
2. Is it ok to delay category/series filtering in OMP for now?
3. Should these filters be added to the submission selection lists (eg - export and catalog entry)? Urgently for 3.1 or as a mid-priority feature?

**Comment by alexxxmendonca** [comment]
Just wondering: is "Review overdue" something we could consider adding to the filters?

It would be a quick way to filter those submissions that have overdue reviews.

**Comment by jmacgreg** [comment]
Hi Nate, 

This looks great! It would be great to add a "Status" filter option as @alexxxmendonca suggests, but if this is a lot of work I think it could be added as a post-3.1 follow-up, so long as the status is indicated in the listing (as it appears to be). It should definitely be considered before too long, though. 

One small note - just confirming that the submission ID will be re-added, correct? 

I have no opinion about the category/series filtering, or adding this to other submission selection lists. The highest priority, which you are achieving here, is the submission dashboard.

**Comment by alexxxmendonca** [comment]
Also, forgot to add, very important: it looks great! Good work, Nate!

**Comment by NateWr** [comment]
@jmacgreg and @alexxxmendonca, see my initial comment in the issue for details about filtering by status. This is highly desired, but exists nowhere in our database, so we can't really query by it. We need to decide and implement some stage status tracking to get there (See: https://github.com/pkp/pkp-lib/issues/1662).

**Comment by jmacgreg** [comment]
Agreed, it looks awesome! And as important, it'll be really useful.

**Comment by NateWr** [comment]
I was able to implement a filter for overdue review assignments. @jmacgreg would you have an opportunity to test this on real data some time before 3.1 goes out? I found a couple bugs where the filter results didn't match the overdue notices in the list. I fixed those, but there may be more and it'd be nice to see how this holds up against a large, active list.

![submissions-list-filter-overdue](https://user-images.githubusercontent.com/2306629/29271720-d752c3a4-80f4-11e7-9a61-14dd512f42c2.gif)

**Comment by alexxxmendonca** [comment]
Excellent! 👍

**Comment by jmacgreg** [comment]
@NateWr yep, can do. Just to confirm, this is all against master?

**Comment by NateWr** [comment]
@jmacgreg yep

**Comment by NateWr** [comment]
@bozana tests passed. Can you code review?

PRs:
https://github.com/pkp/pkp-lib/pull/2704
https://github.com/pkp/ojs/pull/1490
https://github.com/pkp/omp/pull/428

**Comment by bozana** [comment]
👍 Everything merged, thanks!!! Can this issue be closed?

**Comment by NateWr** [comment]
Woohoo! :tada: 

There are additional filtering requests in this issue so I've deferred it for OJS 3.2.

**Comment by jmacgreg** [comment]
Awesome!!!! Thanks Nate, I can’t wait to see this in production.

**Comment by ajnyga** [comment]
Hi @NateWr 

Some journal editors have asked a way to check who is working on which submission. In OJS2 you were apparently able to list submissions by editor, but in OJS3 this is not possible.

So would it be possible to add a "editor filter" here?

**Comment by NateWr** [comment]
@ajnyga It's worth considering. Can you clarify with the editors which need is really important:

- to see ALL submissions assigned to an editor (ie - I want to know what submissions Jane is working on?).
- to see who is assigned to the submissions that are listed (ie - I am interested in this submission. Who is working on it?).

I don't want to have too many filters as the UI will get unwieldy. But we could find other creative ways of doing this in line with #2613. Or including search matches by stage assignments. Or even having data inside the submission be clickable to filter by it.

**Comment by jmacgreg** [comment]
Just a quick note, I was looking at this feature with a client today and we agreed that something like the first option would be nice to have eventually. As an EiC/managing editor, it’s nice to be able to take a look at an individual person’s workload.

**Comment by NateWr** [comment]
:+1: I've updated the original issue.

**Comment by ajnyga** [comment]
Great! It was option 1 they were after, as I suspected.

**Comment by tothbog** [comment]
It looks excellent! :1st_place_medal: As a Journal Manager, what would still add more is a filter to "any kind of overdue". I mean: Editors who take too much time to get reviewers, or even make decisions, reviewers who do not react on the invitation, late with the review, copy editors late with their job, etc. 

I know that too much filters can blow the page, but maybe it should be configurable, as I will never filter by sections or even some of the stages, but this filter I mentioned would be the most used one for sure. 

Anyway, to see the small "alert" icon with the notification what is missing will be already a huge help! :-) Thanks!

**Comment by NateWr** [comment]
@tothbog, I'd love to implement more overdue notices. However, each of the tasks you describe doesn't have a due date in our system, and what's considered "too long" will vary from journal to journal. So we don't have any way to determine if a task is overdue, other than the reviews, where we can check the due dates.

I have some ideas for changing this which I hope to explore in the future, but it requires changes to the workflow, not just the submissions list, so they'll take some thinking.

**Comment by ajnyga** [comment]
Maybe filtering by Issue. See https://forum.pkp.sfu.ca/t/assign-article-to-issues/34079

**Comment by le-ti-ka** [comment]
Hello!
May I ask, if the 20, Oct the actual data of OJS 3.1 release?
and if detailed filtration of submissions by status will make? We really need the submission to be shown in the system in more detail: submission (sent) - submission  (editor is assigned) - review (reviewer is assigned) - review (the first stage) - review (review was received (accepted /  revision request / resubmit for review / resubmit for elsewhere / decline)) - review (complete) - review (2nd stage ...) - review (overdue) - copyediting (accepted) - copyediting  (editor is assigned ) - copyediting  (revision) - copyediting  (complete) - production (accepted for publication) - production (layout editor is assigned ) - production (complete) - published .. something like that.
Thank you beforehand

**Comment by asmecher** [comment]
@le-ti-ka, the current planned release date is Oct. 23, and it looks like we'll make that date. The status information is much richer than it was in previous releases. I'd suggest assessing 3.1 once it's released to see if it meets your needs.

**Comment by NateWr** [comment]
@asmecher Last-minute PR with a fix on some bugs for the overdue filter:

https://github.com/pkp/pkp-lib/pull/2925

TESTS ONLY:
https://github.com/pkp/ojs/pull/1611

**Comment by asmecher** [comment]
Oops -- re-opening as there are still items left.

**Comment by t4x0n** [comment]
Hi, the filter feature is really great but it can be better, of course. I have an idea or request for this filter...  would will be great if we could filter by submission date and last modification date, I was looking and this parameters already exists in the database. This is really important for editors, many times some manuscripts spend more time in the “limbo” because that are forgotten.

**Comment by NateWr** [comment]
Hi @t4x0n,

At the moment, submissions in the lists are ordered by submission date with the newest first. So an editor should be able to locate submissions in limbo by reaching the end of the list and checking the activity logs.

The last modification date probably won't mean what you expect it to mean. It only tracks modifications to the submission object itself (title, etc), but doesn't track activity that happens during the workflow (peer reviews, revisions submitted, etc). A better approach might be to look at the most recent entry in the activity log.

How long something must be inactive before it is considered in limbo will differ for each journal, though, so it's hard to determine this automatically. One possibility would be to have a configurable setting for each journal, and to include submissions inactive for X days in the Overdue filter.

**Comment by le-ti-ka** [comment]
Hi, Nate

We prepared Russian context help for our editors and even switch it on for our OJS 3.1 installation. It works. If it'll be usefull for you, we'll be glad.

Elena Titova
Kazan National Research Technical University, Kazan, Russia

**Comment by NateWr** [comment]
Hi @le-ti-ka, are you talking about the in-context help panel that slides out from the right in OJS? If so, you can contribute a translation to that repository here: https://github.com/pkp/ojs-user-guide

**Comment by jmacgreg** [comment]
Hi Nate (for when you return), 

Revisiting @t4x0n's comments regarding filtering by submission by date - it was put to me this way by a client: 

As an editor, I want to be able to see all submissions that haven't had activity for "X" days, so that I can triage those appropriately. 

"Active" can mean different things, but I think once the statuses per item 3) in the issue description are added, you could do something like: 

- show me all submissions that are overdue by 7 days; 
- show me all submissions that haven't had an editor response in 20 days; 

... and so on, maybe using a slider similar for the # of days as you have for the reviewer lists. 

This is probably a substantial amount of work on top of work that hasn't been done yet, so let me know if it makes sense; and/or if it should be split into a separate bug!

**Comment by NateWr** [comment]
This leans towards the requirement for tasks with due dates in https://github.com/pkp/pkp-lib/issues/1662#issuecomment-322143644. I'm all for it but we'll need due dates attached to tasks before we can make it a reality.

**Comment by NateWr** [comment]
@jmacgreg and @ajnyga: a follow-up on the request to filter by assigned editors. Should this filter be restricted to Journal Managers? Is there a scenario where it's inappropriate for a Subeditor to see what submissions have been assigned to other Subeditors? I noticed that Subeditors do _not_ have access the the All Active or Archives submissions lists.

**Comment by jmacgreg** [comment]
My understanding is that subeditors should only have access to their own submissions, so they probably shouldn't be able to filter by other editors - that should only be available to assigning editors - but I have no super-strong feelings about this.

**Comment by ajnyga** [comment]
I agree with @jmacgreg. Using subeditors is the usual way of handling the author/editor problem in OJS so they should only see submissions assigned to them. 

It is a tool for helping the assignment of submissions anyway so I do not see why subeditors would need it.

**Comment by NateWr** [comment]
Following up on the UI/UX results from the 2018 Sprint, and discussion with the UI/UX team, we decided that quick filters need to be available for editors which breakdown a few different "what i need to do now" tasks:

- In review and no reviewers assigned to current round
- One or more review assignments and acceptance are overdue
- Revisions have been submitted and a new review round has not been started and it has not been advanced to a new stage

**Comment by NateWr** [comment]
We discussed this issue alongside the 2018 Sprint recommendations (#4102) and decided that a high priority would be quick filters that answered "what do I need to do now", broken down by the most common actions. These were:

- In Review stage but no reviewers are assigned to the current round
- In Review stage and one or more reviews is overdue in the current round
- In Review stage and revisions have been submitted for the current round

**Comment by amandastevens** [comment]
Another client has requested the ability to filter the submission list by issue or otherwise see in the submission list (All Active and Archives) what issue a submission is scheduled to be published in or has been published in.

**Comment by NateWr** [comment]
Thanks @amandastevens! Issue assignment should be fairly straightforward because the API already supports getting submissions by issue ID. https://github.com/pkp/ojs/blob/master/classes/services/SubmissionService.inc.php#L49

**Comment by eddoff** [comment]
@NateWr, What is the progress in implementing the issue filter?

**Comment by NateWr** [comment]
Hi @eddoff, there is no progress yet. The REST API supports the filter parameter but we have to build a UI control to select one or more issues to filter by. Since there may be dozens or hundreds of issues, we can not just list them all like other filters.

**Comment by NateWr** [comment]
I believe all of the filters identified here have been implemented or file separately now. Closing this in favor of those issues. If you want to track issues related to submission discovery and tracking, see this project:

https://github.com/pkp/pkp-lib/projects/8
